### ISSUE TITLE
FR: replace() for individual columns

### ISSUE THREAD

**Issue by Peter-Pasta** [h2oai/datatable#3139]
Hi, apologies if I duplicated anything but I didn't find anything like this:

As far as I have seen it is only possible to use replace('a', 'b') for the whole table. Would it be possible to implement this in a way that it can target specific columns or a list of columns.

https://datatable.readthedocs.io/en/latest/api/frame/replace.html?highlight=replace

Thank you guys 👍

**Comment by samukweku** [comment]
Could you kindly provide an example

**Comment by Peter-Pasta** [comment]
Of course 

Example from the docs:

`df.replace(1, -1)`

Extension:

```
df['x'] = f.x.replace(1, -1)
df[:, ['x', 'y']] = dt.replace(1, -1) # if possible
```

**Comment by Peter-Pasta** [comment]
Thought about it and realised this can be done with ifelse.

`df['x'] = dt.ifelse(f.x == 1, -1, f.x)`

Dedicated functionality might make this look more clean but I would withdraw/close my request. 

The only thing that as far as I am aware is not possible is to handle string replace functionality, i.e. replacing characters in a string instead of the complete string itself. Am I right?

**Comment by samukweku** [comment]
You are right @Peter-Pasta at the moment there  is no str replace function yet

**Comment by hughsmith11** [comment]
Being able to replace on a subset of columns would be useful. I have instead had to split the frame into two separate frames, one to do replace on and one I don't want to do replace on. Then cbind those frames back together.

e.g.

```
data = {"A": [1, None, 3, None, 5],
        "B": [4, 5, 6, None, 8],
        "C": [7, 8, 9, 10, 11],
        "D": [5, 7, 2, 9, -1]}

DT = dt.Frame(data)

frame_to_replace = DT[:, "A":"B"]
frame_not_to_replace = DT[:, "C":"D"]

frame_to_replace.replace(None,0)

DT_replaced = dt.cbind(frame_to_replace, frame_not_to_replace)
```

**Comment by samukweku** [comment]
@hughsmith11 , this can be replicated with the `ifelse` function, and a dictionary comprehension:


```py

DT
   |     A      B      C      D
   | int32  int32  int32  int32
-- + -----  -----  -----  -----
 0 |     1      4      7      5
 1 |    NA      5      8      7
 2 |     3      6      9      2
 3 |    NA     NA     10      9
 4 |     5      8     11     -1
[5 rows x 4 columns]


repl = {col : ifelse(f[col] == None, 0, f[col]) for col in ('A','B')}
DT[:, ['A', 'B']] = DT[:, repl]

DT

   |     A      B      C      D
   | int32  int32  int32  int32
-- + -----  -----  -----  -----
 0 |     1      4      7      5
 1 |     0      5      8      7
 2 |     3      6      9      2
 3 |     0      0     10      9
 4 |     5      8     11     -1
[5 rows x 4 columns]
```

abstracting it within `replace` might not be a bad idea, you could possibly raise a PR ( or even abstract this to `fillna`).

**Comment by oleksiyskononenko** [comment]
@hughsmith11 What about doing something like this?

```python
DT_to_replace = DT[:, "A":"B"]
DT_to_replace.replace(None,0)
 DT[:, "A":"B"] = DT_to_replace
```

**Comment by samukweku** [comment]
A dedicated function for forward and backward fill could be helpful

**Comment by samukweku** [comment]
Closed by #3344
